### PR TITLE
Feature/question view swift UI

### PR DESCRIPTION
**Pull request by LorenaL9**
Simulator Screen Shot - iPhone 13 Pro - 2022-09-07 at 15 31 32

### CODE REVIEW

File: QuizApp/Question/View/AnswerViewSwiftUI.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import SwiftUI
+
+struct AnswerViewSwiftUI: View {
```
Comment (marinzg) [inline QuizApp/Question/View/AnswerViewSwiftUI.swift:3]: Why `SwiftUI` suffix?

File: QuizApp/Question/View/Models/QuestionNavigation.swift [hunk 2]
```diff
@@ -1,7 +1,9 @@
+import SwiftUI
+
 struct QuestionNavigation {
 
     let id: Int
-    var correctAnswer: Bool?
+    var correctAnswer: Color
```
Comment (marinzg) [inline QuizApp/Question/View/Models/QuestionNavigation.swift:6]: How can variable named `correctAnswer` be of type `Color`?

File: QuizApp/Question/View/QuestionAnswersView.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        VStack(alignment: .leading, spacing: 38) {
+            Text(question.question)
+                .font(.system(size: 20, weight: .bold))
+
+            VStack(spacing: 16) {
+                ForEach(question.answers, id: \.id) { answer in
+                    AnswerViewSwiftUI(answer: answer)
+                        .onTapGesture {
+                            onAnswered(answer.id)
+                        }
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionAnswersView.swift:20]: Can you call `onTapGesture(perform: onAnswered)`?

File: QuizApp/Question/View/QuestionView.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+                            if self.viewModel.numberOfQuestion > self.viewModel.questions.count {
+                                self.viewModel.showQuizResult()
+                            }
+                        }
+                    }
+                }
+            }
+
+            Spacer()
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionView.swift:57]: You should break this into multiple computed properties.

File: QuizApp/Question/View/QuestionView.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                .frame(height: 5)
+            }
+            .padding(.bottom, 30)
+
+            ForEach(viewModel.questions, id: \.id) { question in
+                let selectedQuestion = viewModel.questionNavigation.filter { questionNavigation in
+                    questionNavigation.isSelected
+                }.first
+
+                if question.id == selectedQuestion?.id {
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionView.swift:40]: Why not `guard`?

File: QuizApp/Question/View/QuestionView.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+                        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
+                            self.viewModel.numberOfSelectedQuestion()
+                            self.viewModel.nextQuestion(isAnswerCorrect: answerId == question.correctAnswerId)
+
+                            if self.viewModel.numberOfQuestion > self.viewModel.questions.count {
+                                self.viewModel.showQuizResult()
+                            }
+                        }
+                    }
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionView.swift:53]: I would prefer if you'd created a function `answerSelected(answerId:_)` and call `QuestionAnswersView(question: question, onAnswered: answerSelected)`.

File: QuizApp/Question/View/QuestionViewModel.swift [hunk 7]
```diff
@@ -36,16 +37,29 @@ class QuestionViewModel {
...
 
+    func markAnswerInQuestionNavigation(isAnswerCorrect: Bool) {
+        let answerBackgroundColor = isAnswerCorrect ? Color.correctAnswer : Color.incorrectAnswer
+
+        questionNavigation = questionNavigation.map { question in
+            if question.isSelected {
+                return QuestionNavigation(id: question.id, correctAnswer: answerBackgroundColor, isSelected: true)
+            } else {
+                return question
+            }
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionViewModel.swift:48]:
```swift
question.isSelected ? 
    QuestionNavigation(id: question.id, correctAnswer: answerBackgroundColor, isSelected: true) : 
    question
```

File: QuizApp/Question/View/QuestionViewModel.swift [hunk 8]
```diff
@@ -65,4 +79,33 @@ class QuestionViewModel {
                 sessionId: sessionId))
     }
 
+    func onAnswered(questionId: Int, answeredId: Int) {
+        questions = questions.map { question in
+            if questionId == question.id {
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionViewModel.swift:84]: This should be a `guard`.

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/Question/View/QuestionAnswersView.swift (hunk 3 above)
Comment (LorenaL9) [reply QuizApp/Question/View/QuestionAnswersView.swift:20]: No😉

File: QuizApp/Question/View/QuestionAnswersView.swift [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import SwiftUI
+
+struct QuestionAnswersView: View {
+
+    @State private var backgroundColor: Color = .lightWhite
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionAnswersView.swift:5]: Are you using this anywhere?

File: QuizApp/Question/View/QuestionViewModel.swift [hunk 10]
```diff
@@ -65,4 +84,50 @@ class QuestionViewModel {
...
+
+    private func numberOfSelectedQuestion() {
+        numberOfQuestion += 1
+    }
+
+    func answerSelected(answerId: Int) {
+        guard let question = questions.filter({ question in
+            question.id == selectedQuestionId
+        }).first
+        else { return }
```
Comment (marinzg) [inline QuizApp/Question/View/QuestionViewModel.swift:118]:
```swift
guard
     let question = questions.filter({ question in question.id == selectedQuestionId}).first
else { return }
```

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/Question/View/QuestionAnswersView.swift (hunk 9 above)
Comment (LorenaL9) [reply QuizApp/Question/View/QuestionAnswersView.swift]: No.

**Review by marinzg — APPROVED** [review]

**Pull request merged**